### PR TITLE
introducing sha-256 checksum support

### PR DESCRIPTION
**Pull request by gaboraranyossy-da**
Implementing #621. This change introduces support for `SHA-256` hashes. The default checksums to use for validation has been changed from `SHA1 -> None` to `SHA-256 -> SHA1 -> None`

### CODE REVIEW

File: tests/shared/src/test/scala/coursier/test/CentralTests.scala [hunk 1]
```diff
@@ -728,6 +728,7 @@ abstract class CentralTests extends TestSuite {
       val ver = "1.17"
 
       def hasSha1(a: Artifact) = a.checksumUrls.contains("SHA-1")
+      def hasSha256(a: Artifact) = a.checksumUrls.contains("SHA-256")
```
Comment (alexarchambault) [inline tests/shared/src/test/scala/coursier/test/CentralTests.scala]: That line isn't necessary any more (since your last commit).

Comment (gaboraranyossy-da) [inline tests/shared/src/test/scala/coursier/test/CentralTests.scala]: Correct, I will remove it soon.

**Comment by gaboraranyossy-da** [comment]
Actually, since I use the library programatically and not in our build system I could just write simply `Cache.file(artifact, checksums = Seq(Some("SHA-256"))) ` at the final step when getting the file.

I don't think using SHA-256 checksums will be standard anytime soon either, so that way I would not need to change the defaults at all.

**Comment by gaboraranyossy-da** [comment]
@alexarchambault would it be possible to merge this master and create a new release?

**Comment by alexarchambault** [comment]
`1.0.0-RC9` will be released once https://github.com/coursier/coursier/pull/627 is merged.

**Comment by gaboraranyossy-da** [comment]
thanks!

**Pull request merged**